src/Rules/Functions/ImplodeFunctionRule.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Php\PhpVersion;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Accessory\AccessoryNonEmptyStringType;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\TypeUtils;
+use PHPStan\Type\UnionType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\FuncCall>
+ */
+class ImplodeFunctionRule implements \PHPStan\Rules\Rule
+{
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof \PhpParser\Node\Name)) {
+			return [];
+		}
+
+		$name = strtolower((string) $node->name);
+		if ($name !== 'implode') {
+			return [];
+		}
+
+		$args = $node->getArgs();
+		if (count($args) !== 2) {
+			return [];
+		}
+
+		$arrayType = $scope->getType($args[1]->value);
+		if ($arrayType->getIterableValueType()->isArray()->yes()) {
```

Comment:
I think the only thing we need to check here is that `$arrayType->getIterableValueType()->toString()` isn't `ErrorType`. No other logic is necessary.

Reply:
I tried
```
		if (!$arrayType->getIterableValueType()->toString() instanceof ErrorType) {
			return [
				RuleErrorBuilder::message(
					sprintf('Call to %s with invalid nested array argument.', $functionName)
				)->build()
			];
		}

```
but this change does not pass the unit tests

Reply:
after re-thinking about your comment, I think I finally understood what you meant.

just adjusted the code to a `toString()` call and also added other invalid non-string test-cases